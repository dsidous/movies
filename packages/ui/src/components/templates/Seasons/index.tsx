import React from 'react';

import Skeleton from './skeleton';
import Season from '../../molecules/Season';
import PageTransition from '../../atoms/PageTransition';
import MiniHeader from '../../molecules/MiniHeader';
import useStyles from './styles';
import { Show } from '@typesRoots/show';

interface Props {
  loading: boolean;
  show: Show;
}

const Seasons: React.FC<Props> = ({
  loading,
  show: { seasons, name, first_air_date, id, poster_path },
}) => {
  const classes = useStyles();

  if (loading) {
    return <Skeleton />;
  }

  return (
    <PageTransition>
      <MiniHeader
        title={name}
        release_date={first_air_date}
        poster_path={poster_path}
        link={`/tv/${id}`}
        linkCopy='Back to main'
      />

      <ul className={classes.list}>
        {seasons
          .sort((a, b) => a.season_number - b.season_number)
          .map((season) => (
            <li className={classes.item} key={season.season_number}>
              <Season season={season} tvId={id} />
            </li>
          ))}
      </ul>
    </PageTransition>
  );
};

export default Seasons;
