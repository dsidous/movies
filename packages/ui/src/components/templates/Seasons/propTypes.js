import { object, bool } from 'prop-types';

export const propTypes = {
  loading: bool,
  show: object,
};

export const defaultProps = {};
