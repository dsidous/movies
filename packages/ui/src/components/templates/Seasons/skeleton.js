import React from 'react';
import Skeleton from '@material-ui/lab/Skeleton';
import { Container, Typography, makeStyles } from '@material-ui/core';

const useStyles = makeStyles(theme => ({
  header: {
    alignItems: 'center',
    borderBottom: theme.separatorBold,
    display: 'flex',
    padding: [[theme.spacing(4), 0]],
  },

  list: {
    padding: 0,
  },

  item: {
    borderBottom: theme.separator,
    display: 'flex',
    justifyContent: 'space-between',
    marginBottom: theme.spacing(2),
    paddingBottom: theme.spacing(2),

    '& h6': {
      marginBottom: theme.spacing(2),
    },
  },
}));

export default () => {
  const classes = useStyles();
  return (
    <Container>
      <Container className={classes.header}>
        <Skeleton variant="rect" width={50} height={80} />
        <Container>
          <Skeleton width="30%" />
          <Skeleton width="20%" />
        </Container>
      </Container>
      <ul className={classes.list}>
        {Array.from(new Array(4)).map((_, index) => (
          <li className={classes.item} key={index}>
            <Skeleton variant="rect" width={300} height={200} />
            <Container>
              <Typography variant="h6">
                <Skeleton variant="text" width="30%" />
              </Typography>
              <Skeleton />
              <Skeleton />
              <Skeleton width="60%" />
            </Container>
          </li>
        ))}
      </ul>
    </Container>
  );
};
