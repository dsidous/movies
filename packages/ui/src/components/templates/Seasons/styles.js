import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(
  theme => ({
    list: {
      margin: '0 auto',
      padding: theme.spacing(0, 3),
    },
    item: {
      borderBottom: theme.separator,
      listStyleType: 'none',
      margin: theme.spacing(5, 0),
      padding: theme.spacing(0, 0, 5),
    },
  }),
  { name: 'seasons' },
);
