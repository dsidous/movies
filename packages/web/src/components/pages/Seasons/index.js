import { compose, withProps } from 'recompose';
import { Seasons } from '@movies/ui';
import { withTv } from '@movies/common';

export default compose(
  withProps(props => ({ tvId: props.match.params.tvId })),
  withTv(),
)(Seasons);
