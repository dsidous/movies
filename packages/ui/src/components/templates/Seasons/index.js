/* eslint-disable camelcase */
import React from 'react';

import { propTypes } from './propTypes';
import Spinner from '../../atoms/Spinner';
import Season from '../../molecules/Season';
import PageTransition from '../../atoms/PageTransition';
import MiniHeader from '../../molecules/MiniHeader';

const Seasons = props => {
  const { loading } = props;

  if (loading) {
    return <Spinner />;
  }

  const {
    show: { seasons, name, first_air_date, id, poster_path },
  } = props;

  return (
    <PageTransition>
      <MiniHeader
        title={name}
        release_date={first_air_date}
        poster_path={poster_path}
        link={`/tv/${id}`}
        linkCopy="Back to main"
      />

      <ul className="seasons">
        {seasons
          .sort((a, b) => a.season_number - b.season_number)
          .map(season => (
            <li className="seasons__element" key={season.season_number}>
              <Season season={season} tvId={id} />
            </li>
          ))}
      </ul>
    </PageTransition>
  );
};

Seasons.propTypes = propTypes;

export default Seasons;
