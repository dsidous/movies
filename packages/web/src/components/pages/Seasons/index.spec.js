import React from 'react';
import { shallow } from 'enzyme';
import Seasons from '.';

const defaultProps = {
  match: { params: { tvId: 123 } },
};

describe('Pages/Seasons', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<Seasons {...defaultProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
