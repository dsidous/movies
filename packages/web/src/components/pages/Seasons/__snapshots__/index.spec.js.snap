// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages/Seasons should render as expected 1`] = `ShallowWrapper {}`;
